code/game/objects/items/weapons/RSF.dm

```diff
@@ -1,116 +1,366 @@
-	// Change mode
+	proc/printsetting()
+		usr << "<span class='notice'>Changed dispensing mode to '[setting]'</span>"
 
-/obj/item/weapon/rsf/afterattack(atom/A, mob/user as mob, proximity)
+	proc/printmessage()
+		usr << "Dispensing [setting]..."
+
+
+	afterattack(atom/A, mob/user as mob, proximity)
+		if(!proximity) return
+		if(disabled && !isrobot(user)) //Disabled RSFs don't work unless you're a borg
+			return 0
+		if (!(istype(A,/obj/structure/table) || istype(A,/turf/simulated/floor) || istype(A,/turf/unsimulated/floor))) //Unsimulated floors are supported if service borgs go on away missions
+			return 0
+
+		switch(mode)
+			if(1)
+				if(useResource(10, user))
+					printmessage()
+					activate()
+					if(istype(A, /turf/simulated/floor))
+						new /obj/item/weapon/spacecash/c10( A )
+					if(istype(A, /obj/structure/table))
+						new /obj/item/weapon/spacecash/c10( A.loc )
+					return 1
+				return 0
+
+			if(2)
+				if(useResource(2, user))
+					printmessage()
+					activate()
+					if(istype(A, /turf/simulated/floor))
+						new /obj/item/weapon/reagent_containers/food/drinks/drinkingglass( A )
+					if(istype(A, /obj/structure/table))
+						new /obj/item/weapon/reagent_containers/food/drinks/drinkingglass( A.loc )
+					return 1
+				return 0
+
+			if(3)
+				if(useResource(1, user))
+					printmessage()
+					activate()
+					if(istype(A, /turf/simulated/floor))
+						new /obj/item/weapon/paper( A )
+					if(istype(A, /obj/structure/table))
+						new /obj/item/weapon/paper( A.loc )
+					return 1
+				return 0
+
+			if(4)
+				if(useResource(2, user))
+					printmessage()
+					activate()
+					if(istype(A, /turf/simulated/floor))
+						new /obj/item/weapon/pen( A )
+					if(istype(A,/obj/structure/table))
+						new /obj/item/weapon/pen( A.loc )
+					return 1
+				return 0
+
+			if(5)
+				if(useResource(10, user))
+					printmessage()
+					activate()
+					if(istype(A, /turf/simulated/floor))
+						new /obj/item/weapon/storage/pill_bottle/dice( A )
+					if(istype(A,/obj/structure/table))
+						new /obj/item/weapon/storage/pill_bottle/dice( A.loc )
+					return 1
+				return 0
+
+			if(6)
+				if(useResource(1, user))
+					printmessage()
+					activate()
+					if(istype(A, /turf/simulated/floor))
+						new /obj/item/clothing/mask/cigarette( A )
+					if(istype(A,/obj/structure/table))
+						new /obj/item/clothing/mask/cigarette( A.loc )
+					return 1
+				return 0
+			else
+				user << "ERROR: RSF in MODE: [mode] attempted use by [user]. Send this text to an admin."
+				return 0
+
+	attack(mob/M as mob, mob/user as mob)
+		if (istype(M, /mob/living/carbon/human) && M != usr) //Only give items to humanoid characters and only if you're giving it to someone else.
+			if (checkResource(1, user)) //Only give the prompt if the RSF has ammo or the borg has battery
+				switch(alert(M,"[usr] wants to print you \a [setting]?",,"Yes","No"))
+					if("Yes")
+						switch(mode)
+							if (1)
+								if(!Adjacent(usr))
```

Comment:
This is the bit that doesn't work. If you can figure out how to make it work, go nuts. It's supposed to check that the two characters are still next to each other but right now it checks if the player is next to itself (I think) and all my attempts to fix it result in always getting false.